### PR TITLE
mrc-360 scoped role widget

### PR DESCRIPTION
**Pull request by hillalex**
Ok, finally! This is the scoped role widget, complete with adding new roles functionality. There's still one piece to go which is implementing the remove functionality (further ticket for this). But this contains commits from all the other open PRs, so need to merge everything else before reviewing this.

### CODE REVIEW

File: src/app/static/src/js/components/reports/reportReadersList.vue [hunk 1]
```diff
@@ -1,14 +1,11 @@
 <template>
-    <div id="report-readers-list">
-        <label class="font-weight-bold">Specific read access</label>
```
Comment (hillalex) [inline src/app/static/src/js/components/reports/reportReadersList.vue]: Moved this label to the view template

File: src/customConfigTests/src/test/kotlin/org/vaccineimpact/orderlyweb/customConfigTests/ReportPagePermissionTests.kt [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+import org.junit.Before
+import org.junit.Test
+import org.openqa.selenium.By
+import org.openqa.selenium.support.ui.ExpectedConditions
+import org.vaccineimpact.orderlyweb.models.Scope
+import org.vaccineimpact.orderlyweb.models.permissions.ReifiedPermission
+import org.vaccineimpact.orderlyweb.test_helpers.giveUserGlobalPermission
+import org.vaccineimpact.orderlyweb.test_helpers.insertReport
+
+class ReportPagePermissionTests : SeleniumTest()
```
Comment (hillalex) [inline src/customConfigTests/src/test/kotlin/org/vaccineimpact/orderlyweb/customConfigTests/ReportPagePermissionTests.kt]: Factored these out into their own class as `ReportPageTests` was getting long and also pulled out shared code into helper methods to make the tests more readable.

**Comment by hillalex** [comment]
Ah, I know what's going on! Only roles with members are shown in the view. We should either
a) change the typeahead to only return roles with members
b) show roles without members

**Comment by EmmaLRussell** [comment]
> Ah, I know what's going on! Only roles with members are shown in the view. We should either
> a) change the typeahead to only return roles with members
> b) show roles without members
> 
> I don't have a strong preference. Maybe the latter?

Yes I think it would be better to include roles without members - clearly it's confusing (to me at least!) to not return the memberless ones. But maybe it should show a hint 'this role currently has no members'

**Comment by hillalex** [comment]
Ok, added roles without members to the query and styled roles without members a little differently, with a `-` instead of an expander icon, similarly to the report table.

**Pull request merged**